tket/src/Placement/GraphPlacement.cpp

```diff
@@ -47,10 +47,11 @@ GraphPlacement::default_pattern_weighting(const Circuit& circuit) const {
        gate_counter < this->maximum_pattern_gates_ && !frontier.slice->empty();
        i++) {
     for (const Vertex& vert : *frontier.slice) {
+      OpType optype = circuit.get_OpType_from_Vertex(vert);
```

Comment:
```suggestion
      if (circuit.get_OpType_from_Vertex(vert) == OpType::Barrier) continue;
```
And revert the changes below? Maybe this is simpler?